crates/core/tedge_mapper/src/c8y/sm_mapper.rs

```diff
@@ -438,7 +450,10 @@ where
     }
 }
 
-async fn execute_operation(payload: &str, command: &str) -> Result<(), SMCumulocityMapperError> {
+pub async fn execute_operation(
+    payload: &str,
+    command: &str,
+) -> Result<(), SMCumulocityMapperError> {
```

Comment:
I really wonder how we can send back the messages expected by Cumulocity to monitor the progress of the operation (`501,Executing`, `502,Failed`, `503,Successful`).

* This can be delegated to the operation plugin.
* This can be delegated to the agent sending operation progress status as currently done for software update.
* Temporarily, this could be done inside the `aync` closure, but this requires an MQTT connection.

The simplest for now is the first option. This is after all what is already done by the Cumulocity operation implementations.

Reply:
Option 1 as it's the simplest to do atm, after all ops handling will be moved to the agent there should be no need for mapper to handle this at all.